packages/utils-debugging-protocol-common/src/RequestResponse.ts

```diff
@@ -0,0 +1,271 @@
+                const encoding = base64Encoded ? 'base64' : 'utf-8';
+                const content = base64Encoded ? atob(body) : body; // There are some JS responses that are base64 encoded for some reason
+
+                rawContent = Buffer.from(body, encoding);
+
+                rBody = {
+                    content,
+                    rawContent,
+                    rawResponse: () => {
+                        return that.getRawResponse();
+                    }
+                };
+            }
+
+            this._response = {
+                body: rBody,
+                charset: null!,
+                headers: normalizedHeaders!,
+                hops: this.hops,
+                mediaType: null!,
+                statusCode: status,
+                url: this.finalUrl
+            };
+
+            const { charset, mediaType } = getContentTypeData(element, this.originalUrl, normalizedHeaders, rawContent);
+
+            this._response.mediaType = mediaType!;
+            this._response.charset = charset!;
+        }
+
+        return this._response;
+    }
+
+    private logInfo(message: string) {
+        debug(`(${this._requestId}) ${message}`);
+    }
+
+    /**
+     * Update the payload received on the `RequestWillBeSent` event for this `requestId`,
+     * taking care of updating other related data if there are any redirects.
+     */
+    public updateRequestWillBeSent(event: Crdp.Network.RequestWillBeSentEvent) {
+        if (event.redirectResponse) {
+            this._hops.push(event.redirectResponse.url);
+        }
+
+        this._willBeSent = event;
+        this.logInfo(RequestStatus.willBeSent);
+    }
+
+    /** Update the payload received on the `ResponseReceived` event for this `requestId`. */
+    public updateResponseReceived(event: Crdp.Network.ResponseReceivedEvent) {
+        this._responseReceived = event;
+        this._status = RequestStatus.responseReceived;
+
+        this.logInfo(RequestStatus.responseReceived);
+    }
+
+    /** Update the payload received on the `LoadingFinished` event for this `requestId`. */
+    public async updateLoadingFinished(event: Crdp.Network.LoadingFinishedEvent) {
+        this._loadingFinished = event;
+        try {
+            this._responseBody = await this._network.getResponseBody!({ requestId: event.requestId });
+        } catch (e) {
+            this.logInfo(`Error getting body`);
+        }
+
+        this._status = RequestStatus.loadingFinished;
+
+        this.logInfo(RequestStatus.loadingFinished);
+    }
+
+    /** Update the payload received on the `LoadingFailed` event for this `requestId`. */
+    public updateLoadingFailed(event: Crdp.Network.LoadingFailedEvent) {
+        this._loadingFailed = event;
+
+        this._status = RequestStatus.loadingFailed;
+        this.logInfo(RequestStatus.loadingFailed);
+    }
+
+    public constructor(network: Crdp.NetworkClient, event: Crdp.Network.RequestWillBeSentEvent, overrideInvalidCert: boolean) {
```

Comment:
IMO, `overrideInvalidCert` should be part of an `options` object. Just in case we need to pass something else in the future.

Reply:
`RequestResponse` is meant to be used only by `utils-debugging-protocol`. If we need to add more things we can then switch to an `options` object without breaking.